examples/xcm/asset_teleportation.ts

```diff
@@ -0,0 +1,67 @@
+import { alice, Rune } from "capi"
+import { signature } from "capi/patterns/signature/polkadot.ts"
+import { types, XcmPallet } from "zombienet/statemine.toml/alice/@latest/mod.js"
+import { chain as parachain, System } from "zombienet/statemine.toml/collator/@latest/mod.js"
+import { Event } from "zombienet/statemine.toml/collator/@latest/types/cumulus_pallet_parachain_system/pallet.js"
+import { RuntimeEvent } from "zombienet/statemine.toml/collator/@latest/types/statemine_runtime.js"
+
+const {
+  VersionedMultiAssets,
+  VersionedMultiLocation,
+  v2: {
+    NetworkId,
+    WeightLimit,
+    junction: { Junction },
+    multilocation: { Junctions },
+    multiasset: { AssetId, Fungibility },
+  },
+} = types.xcm
+
+const aliceBalance = System.Account
+  .value(alice.publicKey)
+  .unhandle(undefined)
+  .access("data", "free")
+
+console.log("Alice balance before:", await aliceBalance.run())
+
+XcmPallet
+  .limitedTeleportAssets({
+    dest: VersionedMultiLocation.V2(Rune.rec({
+      parents: 0,
+      interior: Junctions.X1(Junction.Parachain(1000)),
+    })),
+    beneficiary: VersionedMultiLocation.V2(Rune.rec({
+      parents: 0,
+      interior: Junctions.X1(Junction.AccountId32({
+        id: alice.publicKey,
+        network: NetworkId.Any(),
+      })),
+    })),
+    assets: VersionedMultiAssets.V2(Rune.array([Rune.rec({
+      id: AssetId.Concrete(Rune.rec({
+        parents: 0,
+        interior: Junctions.Here(),
+      })),
+      fun: Fungibility.Fungible(500_000_000_000_000n),
+    })])),
+    feeAssetItem: 0,
+    weightLimit: WeightLimit.Unlimited(),
+  })
+  .signed(signature({ sender: alice }))
+  .sent()
+  .dbgStatus()
+  .finalized()
+  .run()
+
+// TODO: have the recipient associate the downward message with the sender
```

Comment:
@hbulgarini Would you know if there is a standard around this?

Is it possible to correlate a message from the sender to the recipient?

Reply:
not sure specifically what you mean by standard but my recommendation is that for building a teleporting message you can basically copy what the extrinsic `teleport_assets` in the `pallet_xcm` is doing. Specifically this is the sequence of XCM messages built for teleporting: https://github.com/paritytech/polkadot/blob/master/xcm/pallet-xcm/src/lib.rs#L1202:L1207

> Is it possible to correlate a message from the sender to the recipient?

Not AFAIK. The only way that i can think of is setting the recipient as the caller of the extrinsic.